reportportal_client/items/rp_base_item.py

```diff
@@ -0,0 +1,60 @@
+"""
+This module contains functional for Base RP items management.
+
+Copyright (c) 2018 http://reportportal.io .
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from reportportal_client.core.rp_requests import HttpRequest
+
+
+class BaseRPItem(object):
+    """This model stores attributes related to RP item."""
+
+    def __init__(self, rp_url, session, api_version, project_name,
+                 launch_uuid):
+        """Initialize instance attributes.
+
+        :param rp_url:         report portal url
+        :param session:        Session object
+        :param api_version:    RP API version
+        :param project_name:   RP project name
+        :param launch_uuid:    Parent launch UUID
+        """
+        self.uuid = None
+        self.weight = None
+        self.http_requests = list()
+        self._response = None
+        self.rp_url = rp_url
+        self.session = session
+        self.api_version = api_version
+        self.project_name = project_name
+        self.launch_uuid = launch_uuid
+
+    def add_request(self, endpoint, method, request_class, *args, **kwargs):
+        """Add new request object.
+
+        :param endpoint:       request endpoint
+        :param method:         Session object method. Allowable values: get,
+                               post, put, delete
+        :param request_class:  request class object
+        :param args:           request object attributes
+        :param kwargs:         request object named attributes
+        :return: None
+        """
+        http_request = HttpRequest(method, endpoint)
+        rp_request = request_class(*args, **kwargs)
+        rp_request.http_request = http_request
+        rp_request.priority = self.weight
+        self.http_requests.insert(0, rp_request)
```

Comment:
from my point of view we could add request into priority queue (rp wrapper) right here